stglib/iq.py

```diff
@@ -91,142 +96,174 @@ def read_iq(filnam):
 
-def clean_iq(iq):
-    """
-    Preliminary data cleaning when SNR < 0
-    """
-
-    iq["Vel_Mean"].values[iq["Vel_Mean"] < -214748] = np.nan
-    iq["Vel"].values[iq["Vel"] == -214748368] = np.nan
-    for bm in range(4):
-        pr = "Profile_" + str(bm) + "_Vel"
-        iq[pr].values[iq[pr] == -214748368] = np.nan
-        am = "Profile_" + str(bm) + "_Amp"
-        iq[am].values[iq[am] == 65535] = np.nan
-        st = "Profile_" + str(bm) + "_VelStd"
-        iq[st].values[iq[st] < 0] = np.nan
-
-    return iq
-
+    if "equation_velocity_type" in ds.attrs:
+        veltype = {
+            "0": "VelocityXZ (X-Center)",
+            "2": "VelocityXZ (Z-Center)",
+            "3": "VelocityXZ (X-Left)",
+            "4": "VelocityXZ (X-Right)",
+            "5": "Average of all Vx",
+            "10": "Beam 1 Velocity",
+            "11": "Beam 2 Velocity",
+            "12": "Beam 3 Velocity",
+            "13": "Beam 4 Velocity",
+            "14": "VelocityXZ (X-Center, Beam 1 Only)",
+            "15": "VelocityXZ (X-Center, Beam 2 Only)",
+        }
+
+        for v in veltype:
+            if v in ds.attrs["equation_velocity_type"]:
+                ds.attrs["equation_velocity_type"] = veltype[v]
+
+    ds.attrs["survey_point_count"] = int(
+        iqmat["System_IqSetup"]["flowSetup"]["surveyPointCount"]
+    )
+    ds.attrs["survey_point_count"] = int(
+        iqmat["System_IqSetup"]["flowSetup"]["surveyPointCount"]
+    )
+    ds.attrs["IQ_location_Y"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["instrument_Y"] / 1000
+    )
+    ds.attrs["IQ_location_Z"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["instrument_Z"] / 1000
+    )
+    ds.attrs["channel_cross_section_Y"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["channel_Y"][
+            0 : ds.attrs["survey_point_count"]
+        ]
+    ) / 1000
+    ds.attrs["channel_cross_section_Z"] = (
+        iqmat["System_IqSetup"]["flowSetup"]["channel_Z"][
+            0 : ds.attrs["survey_point_count"]
+        ]
+    ) / 1000
+
+    ds.attrs[
+        "channel_cross_section_note"
+    ] = "Y = distance (m) along transect across the channel, starting on right bank. Z = elevation (m) of bed referenced to geopotential datum. Measurements collected in the field at the beginning of each IQ deployment."
+    ds.attrs[
+        "IQ_location_note"
+    ] = "Location of IQ projected onto cross channel transect. Actual location of IQ may be up to 2 meters off the across-channel transect (in perpendicular direction or along channel)."
 
-def vel_to_ms(iq):
-    """
-    Convert velocity data from mm/s to m/s
-    """
+    for k in iqmat["System_IqSetup"]["basicSetup"]:
+        if "spare" not in k:
+            ds.attrs[k] = iqmat["System_IqSetup"]["basicSetup"][k]
+    for k in iqmat["System_Id"]:
+        ds.attrs[k] = iqmat["System_Id"][k]
 
-    for var in ["FlowData_Vel_Mean", "FlowData_Vel"]:
-        iq[var] = iq[var] / 1000
+    if ds.attrs["InstrumentType"] == "IQ":
```

Comment:
Is InstrumentType provided in the mat file? Or is this something the user adds?

Reply:
This comes from the mat file. There is another IQ called IQ Pipe and the transducers are at different angles. We do not have an IQ Pipe, but I was assuming the instrument type would say 'IQ Pipe" or something different than just 'IQ' in the mat file. I can take that if statement out and just standardize the beam angle if we don't think any offices will be using an IQ Pipe? As the name suggests, the IQ Pipe is meant for closed conduits.